__tests__/app.test.js

```diff
@@ -0,0 +1,46 @@
+const app = require('../db/app');
+const db = require('../db/connection');
+
+const request = require('supertest');
+const seed = require('../db/seeds/seed.js');
+const {articleData, commentData, topicData, userData} = require('../db/data/test-data/index');
+const {getTopics} = require('../controllers/controllers.js');
+
+
+
+beforeEach(() =>{
+    return seed({ articleData, commentData, topicData, userData })
+    })
+afterAll(() => {
+    db.end()
+})
+
+describe('General testing', () => {
+    test('should return 404 if path spelt wrong or does not exist', () =>{
+        return request(app)
+        .get('/api/not-existing-path')
+        .expect(404)
+        .then((response) => {
+            expect(response.body.message).toBe('path is not found')
+        })
+    });
+})
+
+describe.only('GET /api/topics', () =>{
+    test('responds with status 200', () =>{
+        return request(app)
+        .get('/api/topics')
+        .expect(200)
+    });
+
+    test('returns array of topics', () =>{
+        return request(app)
+        .get('/api/topics')
+        .then(({body}) => {
+            body.topics.forEach((topic) => {
```

Comment:
great job on iterating through and and checking each element in the array.
However when we are testing we should do a length check on the array before a forEach, an empty array would cause this to be a false positive.